### ISSUE TITLE
integer overflow in cliprdr_read_format_list function

### ISSUE THREAD

**Issue by Kcchouette** [FreeRDP/FreeRDP#6092]
**Describe the bug**

I think I've found a integer overflow in cliprdr_read_format_list function https://github.com/FreeRDP/FreeRDP/blob/511f89cd99b60c5e755d5030c86df0d673a756de/channels/cliprdr/cliprdr_common.c#L499-L515 (imagine `dataLen` is never equal to 0...)

**To Reproduce**

When we have a malformed stream, we have an integer overlfow in the function above.

**Additional context**

Here a quick & dirty fix:

```
diff --git a/channels/cliprdr/cliprdr_common.c b/channels/cliprdr/cliprdr_common.c
index ebe0d6aad..9bef70150 100644
--- a/channels/cliprdr/cliprdr_common.c
+++ b/channels/cliprdr/cliprdr_common.c
@@ -511,6 +511,14 @@ UINT cliprdr_read_format_list(wStream* s, CLIPRDR_FORMAT_LIST* formatList, BOOL
 			Stream_Seek(s, (formatNameLength + 1) * 2);
 			dataLen -= ((formatNameLength + 1) * 2);
 
+			/* Overflow, stream is malformed */
+			if (formatList.dataLen <= dataLen)
+			{
+				WLog_ERR(TAG, "Could not parse cliport stream");
+				error = CHANNEL_RC_BAD_PROC;
+				goto error_out;
+			}
+
 			formatList->numFormats++;
 		}
 

```

**Comment by akallabeth** [comment]
Fixed with #6151 and #6152

**Comment by Kcchouette** [comment]
(seems fixed in commit b230ac98e9e5f0c6b05b3b7f4f14016977bb5395)
